tests/testthat/test-valid.R

```diff
@@ -38,6 +38,26 @@ test_that("validate checks base type", {
   expect_snapshot(error = TRUE, validate(x))
 })
 
+test_that("validate checks the type of setters", {
+  foo <- new_class("foo", properties = list(x =
+    new_property(
+      class_double,
+      setter = function(self, value) {
+        self@x <- as.character(value)
+        self
+      }
+    )
+  ))
+  expect_snapshot(foo(x = 123), error = TRUE)
+})
+
+test_that("validate does not check type of getters", {
```

Comment:
Maybe a note about why? i.e. something like "because getters can be arbitrarily expensive, so we require the class author to uphold this"